### PR TITLE
[Dygraph]add export_model and deploy

### PR DESCRIPTION
**Pull request by jerrywgz**

### CODE REVIEW

File: deploy/cpp/include/object_detector.h [hunk 1]
```diff
@@ -88,12 +97,13 @@ class ObjectDetector {
       const cv::Mat& raw_mat,
       std::vector<ObjectResult>* result);
 
-  std::unique_ptr<paddle::PaddlePredictor> predictor_;
+  std::shared_ptr<Predictor> predictor_;
```
Comment (wangxinxin08) [inline deploy/cpp/include/object_detector.h:100]: why shared_ptr?

File: deploy/python/infer.py [hunk 2]
```diff
@@ -536,33 +343,67 @@ def predict_video():
...
+
 if __name__ == '__main__':
+    paddle.enable_static()
     parser = argparse.ArgumentParser(description=__doc__)
     parser.add_argument(
         "--model_dir",
         type=str,
         default=None,
-        help=("Directory include:'__model__', '__params__', "
+        help=("Directory include:'model.pdiparams', 'model.pdmodel', "
```
Comment (yghstill) [inline deploy/python/infer.py:380]: model.pdiparams  ->  model.pdparams

File: deploy/cpp/include/object_detector.h (hunk 1 above)
Comment (jerrywgz) [reply deploy/cpp/include/object_detector.h:100]: the change due to the upgrade of Predictor in Paddle Inference

File: deploy/cpp/CMakeLists.txt [hunk 3]
```diff
@@ -112,8 +113,9 @@ endif()
 
 if (NOT WIN32)
   if (WITH_TENSORRT AND WITH_GPU)
-     include_directories("${TENSORRT_DIR}/include")
-     link_directories("${TENSORRT_DIR}/lib")
+	  include_directories("${TENSORRT_INC_DIR}/include")
```
[the remark this answers is not in the document]
Comment (jerrywgz) [reply deploy/cpp/CMakeLists.txt]: Done, thx and actually this CMakeList is the same as master branch

File: deploy/cpp/docs/linux_build.md [hunk 4]
```diff
@@ -40,38 +40,48 @@ fluid_inference
...
+CUDA_LIB=/path/to/cuda/lib
+
 # CUDNN 的 lib 路径
-CUDNN_LIB=/path/to/cudnn/lib/
+CUDNN_LIB=/path/to/cudnn/lib
 
-# OPENCV 路径, 如果使用自带预编译版本可不修改
-sh $(pwd)/scripts/bootstrap.sh  # 下载预编译版本的opencv
-OPENCV_DIR=$(pwd)/deps/opencv3gcc4.8/
+修改脚本设置好主要参数后，执行`build`脚本：
```
[the remark this answers is not in the document]
Comment (jerrywgz) [reply deploy/cpp/docs/linux_build.md]: done, thx

File: deploy/cpp/docs/linux_build.md [hunk 5]
```diff
@@ -40,38 +40,48 @@ fluid_inference
...
+```shell
+sh ./scripts/build.sh
+```
+
+# 请检查以上各个路径是否正确
```
[the remark this answers is not in the document]
Comment (jerrywgz) [reply deploy/cpp/docs/linux_build.md:71]: done, thx

File: deploy/cpp/docs/linux_build.md [hunk 6]
```diff
@@ -85,19 +95,23 @@ make
  ```shell
```
[the remark this answers is not in the document]
Comment (jerrywgz) [reply deploy/cpp/docs/linux_build.md]: done, thx

File: deploy/python/infer.py (hunk 2 above)
Comment (jerrywgz) [reply deploy/python/infer.py:380]: the output formats of jit.save are pdiparams. pdiparams.info and pdmodel

**Review by wangxinxin08 — COMMENTED** [review]
LGTM

**Review by wangxinxin08 — APPROVED** [review]

File: deploy/python/infer.py [hunk 7]
```diff
@@ -321,16 +233,17 @@ def load_predictor(model_dir,
...
+        'trt_fp32': Config.Precision.Float32,
+        'trt_fp16': Config.Precision.Half
     }
-    config = fluid.core.AnalysisConfig(
-        os.path.join(model_dir, '__model__'),
-        os.path.join(model_dir, '__params__'))
     if use_gpu:
         # initial GPU memory(M), device ID
-        config.enable_use_gpu(100, 0)
+        config.enable_use_gpu(1000, 0)
```
Comment (qingqing01) [inline deploy/python/infer.py:246]: 1000M, 1G有点大

File: deploy/python/preprocess.py [hunk 8]
```diff
@@ -0,0 +1,216 @@
...
+
+from PIL import Image
+import cv2
+import numpy as np
+
+
+def decode_image(im_file, im_info):
+    """read rgb image
+    Args:
+        im_file (str/np.ndarray): path of image/ np.ndarray read by cv2
```
Comment (qingqing01) [inline deploy/python/preprocess.py:23]: str|np.ndarray

input can be image path or np.ndarray

File: ppdet/utils/export_utils.py [hunk 9]
```diff
@@ -0,0 +1,120 @@
+# Copyright (c) 2020 PaddlePaddle Authors. All Rights Reserved.
```
Comment (qingqing01) [inline ppdet/utils/export_utils.py:1]: 只有tools/export_model.py用到的文件，就只放到tools下面吧

File: deploy/python/preprocess.py (hunk 8 above)
Comment (jerrywgz) [reply deploy/python/preprocess.py]: Done, thx

File: ppdet/utils/export_utils.py (hunk 9 above)
Comment (jerrywgz) [reply ppdet/utils/export_utils.py]: Done, thx

File: deploy/python/infer.py (hunk 7 above)
Comment (jerrywgz) [reply deploy/python/infer.py]: Done, thx

**Review by yghstill — APPROVED** [review]

**Review by yghstill — APPROVED** [review]

**Pull request merged**